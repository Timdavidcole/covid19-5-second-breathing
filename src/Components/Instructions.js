import React from "react";
import "../Styles/Instructions.css";

export default class Instructions extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      interval: 4,
      getReady: true
    };

    this.instructionChooser = this.instructionChooser.bind(this);
    this.timer = this.timer.bind(this);
    this.start = this.start.bind(this);
  }

  componentDidMount() {
    setTimeout(this.start, 12000);
    setTimeout(this.timer, 12000);
  }

  start() {
    this.setState({ getReady: false })
  }

  timer() {
    setInterval(() => {
      const { interval } = this.state;
      if (interval > 0) {
        this.setState(({ interval }) => ({
          interval: interval - 1
        }));
      }
      if (interval === 1) {
        this.setState(() => ({
          interval: 4
        }));
      }
    }, 5000);
  }

  componentWillUnmount() {
    clearInterval(this.myInterval);
  }

  instructionChooser() {
    const { interval, getReady } = this.state;
    if (getReady) {
      return <h2>Get ready...</h2>;
    } else if (interval === 4) {
      return <h2>Breathe in</h2>;
    } else if (interval === 2) {
      return <h2>Breathe out</h2>;
    } else {
      return <h2>Hold</h2>;
    }
  }

  render() {
    return <div className="instructions">{this.instructionChooser()} </div>;
  }
}
