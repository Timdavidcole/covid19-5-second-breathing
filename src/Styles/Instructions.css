.instructions {
    position: absolute;
    top: 10%;
    width: 100%;
    margin: 5px;
}

.instructions h2 {
    font-size: 35px;
    text-align: center;
    color: #34d0eb;
    animation: text-color 16s infinite, fade-in-fade-out 8s infinite;
    text-shadow: 0px 0px 7px rgba(255, 255, 255, 0.543);
}

@keyframes text-color {
    0% {
        color: #34d0eb;
        text-shadow: 0px 0px 0px rgba(255, 255, 255, 0.687);
    }

    25% {
        color: #2757a4;
        text-shadow: 0px 4px 6px rgba(108, 108, 108, 0.77);
    }

    50% {
        color: #2757a4;
        text-shadow: 0px 4px 6px rgba(108, 108, 108, 0.77);
    }

    75% {
        color: #34d0eb;
        text-shadow: 0px 0px 0px rgba(255, 255, 255, 0.687);
    }

    100% {
        color: #34d0eb;
        text-shadow: 0px 0px 0px rgba(255, 255, 255, 0.687);
    }
}

@keyframes fade-in-fade-out {
    0%   { opacity:0; }
    10%  { opacity:1; }
    90%  { opacity:1; }
    100% { opacity:0; }
  }