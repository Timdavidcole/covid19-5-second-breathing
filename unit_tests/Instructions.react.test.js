import React from 'react';
import Instructions from '../src/Components/Instructions';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Instructions />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});