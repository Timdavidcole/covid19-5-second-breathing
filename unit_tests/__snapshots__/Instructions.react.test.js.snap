// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="instructions"
>
  <h2>
    Get ready...
  </h2>
   
</div>
`;
